index.js

```diff
@@ -551,6 +552,19 @@ Browserify.prototype._createDeps = function (opts) {
                 return through();
             }
         }
+
+        if (opts.commondir === false && opts.builtins === false) {
+          opts.insertGlobalVars = xtend({
+            __dirname: function(file, basedir) {
+              var dir = path.dirname(path.relative(basedir, file));
+              return 'require("path").join(__dirname,"' + dir.split(path.sep).join('","') + '")';
```

Comment:
paths _can_ contain " on some filesystems, could you use JSON.stringify here instead of manually wrapping the path in quotes?

Reply:
Sounds good, done!